File: backend/src/main/java/com/bms/backend/global/error/ApiProblem.java
package com.bms.backend.global.error;

import java.util.List;

public record ApiProblem(
        String code, String message, String traceId, List<ApiFieldError> fieldErrors) {

    public ApiProblem {
        fieldErrors = fieldErrors == null ? List.of() : List.copyOf(fieldErrors);
    }
}

File: frontend/shared/types/api-problem.ts
export type ApiFieldErrorReason =
  | "REQUIRED"
  | "SIZE"
  | "FORMAT"
  | "INVALID";

export type ApiFieldError = {
  field: string;
  reason: ApiFieldErrorReason;
  message: string;
};

export type ApiProblem = {
  code: string;
  message: string;
  traceId: string;
  fieldErrors: ApiFieldError[];
};
